lib/rspec-puppet-facts.rb

```diff
@@ -72,14 +61,14 @@ def on_supported_os(opts = {})
                                     "/^#{operatingsystemmajrelease}-/"
                                   end
             elsif os_sup['operatingsystem'] =~ /Windows/i
-              hardwaremodel = version =~ /^[12]\./ ? 'x64' : 'x86_64'
+              hardwaremodel = facterversion =~ /^[12]\./ ? 'x64' : 'x86_64'
               os_sup['operatingsystem'] = os_sup['operatingsystem'].downcase
               operatingsystemmajrelease = operatingsystemmajrelease[/\A(?:Server )?(.+)/i, 1]
 
               # force quoting because windows releases can contain spaces
               os_release_filter = "\"#{operatingsystemmajrelease}\""
 
-              if operatingsystemmajrelease == '2016' && Puppet::Util::Package.versioncmp(version, '3.4') < 0
+              if operatingsystemmajrelease == '2016' && Puppet::Util::Package.versioncmp(facterversion, '3.4') < 0
```

Comment:
Here I have my doubts because `facterversion` may be 3.4.0 but if there are no 3.4.0 facts it can fall back to 3.3.x.